app/repository.go

```diff
@@ -3,5 +3,5 @@ package app
 import "golang_template/internal/repositories"
 
 func (a *application) InitRepositories() repositories.UserRepository {
-	return repositories.NewUserRepository()
+	return repositories.NewUserRepository(a.InitDatabase(), &a.ctx)
```

Comment:
The database must be injected into the repository. In this case, setting up the database again is not the right thing to do. So we must include the database in the repository setup entry.
Like this:

```
func (a *application) InitRepositories(db database.Database) repositories.UserRepository {
	return repositories.NewUserRepository(db, &a.ctx)
}
```